### PR TITLE
canonical/multipass dynamic inventory

### PR DESCRIPTION
**Pull request by scodeman**
##### SUMMARY
Dynamic inventory for [canonical/multipass](https://github.com/canonical/multipass)

##### ISSUE TYPE
- Feature Pull Request
##### COMPONENT NAME
multipass

### CODE REVIEW

**Comment by scodeman** [comment]
> Sorry, we no longer accept new inventory scripts. Please create an [inventory plugin](https://docs.ansible.com/ansible/2.10/plugins/inventory.html) instead.

Just committed a plugin for it

File: plugins/inventory/multipass.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
+# Copyright (C) 2020  Florent Madiot <scodeman@scode.io>
+# Copyright (c) 2020 Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = '''
+    name: multipass
+    plugin_type: inventory
```
Comment (felixfontein) [inline plugins/inventory/multipass.py:10]:
```suggestion
    plugin_type: inventory
    author: ... please fill in ...
    version_added: 1.2.0
```

File: plugins/inventory/multipass.py [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+    short_description: canonical/multipass inventory source
+    description:
+        - Get inventory hosts from the local multipass installation.
+        - Uses a YAML configuration file with a valid YAML extension.
+        - The inventory_hostname is always the 'Name' of the multipass instance.
```
Comment (felixfontein) [inline plugins/inventory/multipass.py:15]:
```suggestion
        - The C(inventory_hostname) is always the 'Name' of the multipass instance.
```

File: plugins/inventory/multipass.py [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+    extends_documentation_fragment:
+      - constructed
```
Comment (felixfontein) [inline plugins/inventory/multipass.py:17]:
```suggestion
      - ansible.builtin.constructed
```

File: plugins/inventory/multipass.py [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+      - inventory_cache
```
Comment (felixfontein) [inline plugins/inventory/multipass.py:18]:
```suggestion
      - ansible.builtin.inventory_cache
```

File: plugins/inventory/multipass.py [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+    requirements:
+      - multipass CLI installed
+    options:
+        plugin:
+            description: token that ensures this is a source file for the 'multipass' plugin.
+            required: True
+            choices: ['multipass', 'community.general.multipass']
```
Comment (felixfontein) [inline plugins/inventory/multipass.py:25]:
```suggestion
            choices: [ community.general.multipass ]
```

File: plugins/inventory/multipass.py [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+        - Uses a YAML configuration file with a valid YAML extension.
+        - The inventory_hostname is always the 'Name' of the multipass instance.
+    extends_documentation_fragment:
+      - constructed
+      - inventory_cache
+    requirements:
+      - multipass CLI installed
+    options:
+        plugin:
+            description: token that ensures this is a source file for the 'multipass' plugin.
```
Comment (felixfontein) [inline plugins/inventory/multipass.py:23]:
```suggestion
            description: Token that ensures this is a source file for the C(community.general.multipass) plugin.
```

File: plugins/inventory/multipass.py [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+            required: True
```
Comment (felixfontein) [inline plugins/inventory/multipass.py:24]:
```suggestion
            required: true
```

File: plugins/inventory/multipass.py [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+            choices: ['multipass', 'community.general.multipass']
+        running_only:
+            description: toggles showing all vms vs only those currently running
```
Comment (felixfontein) [inline plugins/inventory/multipass.py:27]:
```suggestion
            description: Toggles showing all VMs vs only those currently running.
```

File: plugins/inventory/multipass.py [hunk 9]
```diff
@@ -0,0 +1,120 @@
...
+            type: boolean
+            default: False
```
Comment (felixfontein) [inline plugins/inventory/multipass.py:29]:
```suggestion
            default: false
```

File: scripts/inventory/multipass.py [hunk 10]
```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env python
+"""
```
Comment (felixfontein) [inline scripts/inventory/multipass.py:2]: Please remove the script.

**Review by felixfontein — COMMENTED** [review]
You need some kind of tests for this plugin. You could do integration tests with a mock shell script named `multipass` which returns some fake data.

File: plugins/inventory/multipass.py (hunk 1 above)
Comment (scodeman) [reply plugins/inventory/multipass.py:10]:
```suggestion
    plugin_type: inventory
    author: 
      - Florent Madiot (@scodeman)
    version_added: 1.2.0
```

**Comment by scodeman** [comment]
> You need some kind of tests for this plugin. You could do integration tests with a mock shell script named `multipass` which returns some fake data.

Ok I will look at that

File: plugins/inventory/multipass.py [hunk 11]
```diff
@@ -0,0 +1,123 @@
...
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = '''
+    name: multipass
+    plugin_type: inventory
+    author:
+      - Florent Madiot (@scodeman)
+    version_added: 1.2.0
```
Comment (Andersson007) [inline plugins/inventory/multipass.py:13]:
```suggestion
    version_added: 1.3.0
```

**Comment by Andersson007** [comment]
@scodeman hi, how about the tests?

File: tests/integration/targets/inventory_multipass/aliases [hunk 12]
```diff
@@ -0,0 +1,8 @@
+disabled
```
Comment (felixfontein) [inline tests/integration/targets/inventory_multipass/aliases:1]: Have you tried whether it actually works in CI?

**Comment by felixfontein** [comment]
@scodeman are you still working on this?

needs_info

**Comment by felixfontein** [comment]
Closing since there was no more reaction. @scodeman if you're interested again in working on this, please ping me. If someone else is interested in continuing this PR, please also write it here so we can coordinate.

**Pull request closed without merging**